### PR TITLE
Feat data item customization for v2 data items

### LINKED ISSUE #3427: Downsample Method for dashboard editor
Downsample Method for dashboard editor 

### What package is this for?

- [x] React
- [ ] Angular

### Summary
Create new dropdown for V2 data items in dashboard editor 

More details can be found in https://github.ibm.com/wiotp/Maximo-Asset-Monitor/issues/3955

### PR DESCRIPTION
**Pull request by EdmondCShaffer**
Closes #3427
Related to https://github.ibm.com/wiotp/Maximo-Asset-Monitor/issues/3955

**Summary**
New dropdown for V2 data items in dashboard editor

**Change List (commits, features, bugs, etc)**
DataSeriesFormItemModal 

**Acceptance Test (how to verify the PR)**
when data item has downsample methods the downsample drop down should be shown when editing 

**Regression Test (how to make sure this PR doesn't break old functionality)**

- tests here

**Things to look for during review**

- [ ] Make sure all references to `iot` or `bx` class prefix is using the prefix variable
- [ ] (React) All major areas have a `data-testid` attribute. New test ids should have test written to ensure they are not changed or removed.
- [ ] UI should be checked in RTL mode to ensure the proper handling of layout and text.
- [ ] All strings should be translatable.
- [ ] The code should pass a11y scans (The storybook a11y knob should show no violations). New components should have a11y test written.
- [ ] Unit test should be written and should have a coverage of 90% or higher in all areas.
- [ ] All components should be passing visual regression test. For new styles or components either a visual regression test should be written for all permutations or the base image updated.
- [ ] Changes or new components should either write new or update existing documentation.
- [ ] PR should link and close out an existing issue

### CODE REVIEW

File: packages/react/src/components/CardEditor/CardEditForm/CardEditFormItems/DataSeriesFormItemModal.jsx [hunk 1]
```diff
@@ -78,6 +78,13 @@ const propTypes = {
       })
     ),
     aggregationMethod: PropTypes.string,
+    downSampleMethods: PropTypes.arrayOf(
+      PropTypes.shape({
+        id: PropTypes.string,
+        text: PropTypes.string,
+      })
+    ),
+    downSampleMethod: PropTypes.string,
```
Comment (davidicus) [inline packages/react/src/components/CardEditor/CardEditForm/CardEditFormItems/DataSeriesFormItemModal.jsx:87]: should this be part of the `i18n` prop definition?

**Review by davidicus — COMMENTED** [review]
This looks pretty good. There is just the issue with the sting prop type not being in the `i18n` object prop definition.

File: packages/react/src/components/CardEditor/CardEditForm/CardEditFormItems/DataSeriesFormItemModal.jsx (hunk 1 above)
Comment (EdmondCShaffer) [reply packages/react/src/components/CardEditor/CardEditForm/CardEditFormItems/DataSeriesFormItemModal.jsx:87]: Downsample methods follow the same structure as Aggregation methods. We handle the translation in monitor.

**Review by davidicus — APPROVED** [review]

**Comment by davidicus** [comment]
I think you may need an update from `next`. I am unable to push to your branch but I was opening up a PR to fix these snapshots that you could then merge in and it looks like they have already been updated in `next`.

**Comment by herleraja** [comment]
I tried to link it with monitor UI and seems be no cards are throwing following issues. 
image

```
Uncaught TypeError: cardRef is not a function
    at ref (Card.js?c82c:439:1)
```

@kevinsperrine I see that you have changed some card level codes. Maybe did that caused these issues? 
image

**Comment by davidicus** [comment]
Errors reported stem from linking project locally with mismatched yarn versions.

**Pull request merged**